greymatter-io/terraform-greymatter-ecs
# Private subnets & ssl

This PR adds private subnets to the main setup (necessary in order to use the [`awsvpc` networking type](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-networking.html) for the ecs tasks). Removes all load balancers in favor of service discovery using a route53 hosted zone and updates the control/sidecar env vars for this, and adds certs to the docker containers and turns on TLS.

Closes #16  , #6 

To test,

1. Follow the instructions in the readme to generate those parameters/secrets manually and grab their arns. Checkout this branch, `private_subnets`.

2. Create a test.tfvars file that looks like this:

```
docker_secret_arn = 
access_key_arn = 
secret_access_key_arn = 
aws_region = 
key_pair_name = 
```

This should be all you need to get this set up with defaults.  

3. Change [this line](https://github.com/greymatter-io/terraform-greymatter-ecs/blob/2c191da1e88b49f8a70f4716d8195471b5b615f6/main.tf#L86) to use `source = "./greymatter"` and `terraform init`.

4. Run `terraform apply -var-file=test.tfvars`.

Everything should come up without errors, verify in the ecs console that all 4 services are running (control may restart a couple of times waiting for control-api). In the cloudwatch console, check the logs for control/control-api. Make sure the control logs get past the connection to control-api.  The sidecar logs will show repeating `REST update for /v2/discovery:listeners failed` this doesnt mean that the connection has failed - it just means that there are no mesh configs for it to discover from control yet.
